index.js

```diff
@@ -1,159 +1,88 @@
 'use strict';
-var execFile = require('child_process').execFile;
-var which = require('which');
-var eachAsync = require('each-async');
-var semver = require('semver');
-var userHome = require('user-home');
-var latestVersion = require('latest-version');
-var sortOn = require('sort-on');
-var checks = [];
-
-function binaryCheck(bin, opts, cb) {
+const which = require('which');
+const semver = require('semver');
+const userHome = require('user-home');
+const latestVersion = require('latest-version');
+const sortOn = require('sort-on');
+const execa = require('execa');
+const pify = require('pify');
+
+const whichP = pify(which);
+
+const binaryCheck = (bin, opts) => {
 	opts = opts || {};
 
-	var title = opts.title || bin[0].toUpperCase() + bin.slice(1);
+	const title = opts.title || bin[0].toUpperCase() + bin.slice(1);
 
-	which(bin, function (err) {
-		if (err) {
+	return whichP(bin)
+		.then(() => ({title, fail: false}))
+		.catch(err => {
 			if (/not found/.test(err.message)) {
-				cb(null, {
-					title: title,
+				return {
+					title,
 					message: opts.message || 'Not installed',
 					fail: true
-				});
-				return;
+				};
 			}
 
-			cb(err);
-			return;
-		}
-
-		cb(null, {title: title});
-	});
-}
-
-[
-	'ruby',
-	'compass',
-	'git'
-].forEach(function (el) {
-	checks.push(function binaries(cb) {
-		binaryCheck(el, null, cb);
-	});
-});
-
-checks.push(function home(cb) {
-	cb(null, {
-		title: process.platform === 'win32' ? '%USERPROFILE' : '$HOME',
-		message: !userHome && [
-			'environment variable is not set. This is required to know where',
-			'your home directory is. Follow this guide:',
-			'https://github.com/sindresorhus/guides/blob/master/set-environment-variables.md'
-		].join(' '),
-		fail: !userHome
-	});
-});
-
-checks.push(function node(cb) {
-	execFile(process.execPath, ['--version'], function (err, stdout) {
-		if (err) {
-			cb(err);
-			return;
-		}
-
-		var version = stdout.trim();
-		var pass = semver.satisfies(version, '>=0.10.0');
-
-		cb(null, {
-			title: 'Node.js',
-			message: !pass && version + ' is outdated. Please update: http://nodejs.org',
-			fail: !pass
+			throw err;
 		});
-	});
-});
-
-checks.push(function npm(cb) {
-	var bin;
-
-	try {
-		bin = which.sync('npm');
-	} catch (err) {
-		cb(null, {
-			title: 'npm',
-			message: 'Not installed. Please install Node.js (which bundles npm) from http://nodejs.org',
-			fail: true
-		});
-		return;
-	}
-
-	execFile(bin, ['--version'], function (err, stdout) {
-		if (err) {
-			cb(err);
-			return;
-		}
+};
 
-		var localVersion = stdout.trim();
+const binaryVersionCheck = bin => {
+	const title = bin;
+	const message = `Not installed. Please install it by running: npm install -g ${bin}`;
 
-		latestVersion('npm').then(function (version) {
-			var pass = semver.satisfies(localVersion, version);
+	return execa.stdout(bin, ['--version'])
```

Comment:
Because I pushed my changes it removed my inline message. So I'll add it again :).

I removed the `whichP` call here to test if `bin` exists because `execa` will fill with `spawn yo ENOENT` if `yo` could not be found. So actually with `execa.stdout` we can catch that as well by testing `if (/ENOENT/.test(err.message))` below.

Is this ok or am I missing something and is `which` really necessary here?